apps/src/lib/levelbuilder/lesson-editor/AddLevelFilters.jsx

```diff
@@ -13,11 +14,43 @@ const styles = {
   }
 };
 
-//TODO Hook up the filtering system to work on the real levels
-//Selects need real data added into them
-
 export default class AddLevelFilters extends Component {
-  static propTypes = {};
+  static propTypes = {
+    searchFields: PropTypes.object,
+    handleSearch: PropTypes.func
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      levelName: '',
+      levelType: this.props.searchFields.levelOptions[0][1],
+      scriptId: this.props.searchFields.scriptOptions[0][1],
+      ownerId: this.props.searchFields.ownerOptions[0][1]
+    };
+  }
+
+  handleSearch = () => {
+    const {levelName, levelType, scriptId, ownerId} = this.state;
+    this.props.handleSearch(levelName, levelType, scriptId, ownerId);
```

Comment:
because this is an expensive call to the database, it could be a good idea to "debounce" it here so that we do not do this too often. https://github.com/code-dot-org/code-dot-org/blob/84faca566fc766e1166cd5bc1a0d65e407c74164/apps/src/StudioApp.js#L304-L307
```
handleSearch = _.debounce(() => {
  ...
}, 1000);
```
I guess another option would be to disable the button while the request is in progress, but this seems like an easier/quicker approach.